### PR TITLE
libsmacker: add libsmacker recipe

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **libsmacker/1.1.1r35**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libsmacker/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
+from conans import AutoToolsBuildEnvironment, ConanFile, tools
+from contextlib import contextmanager
+import os
+
+
+class LibsmackerConan(ConanFile):
+    name = "libsmacker"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "http://libsmacker.sourceforge.net/#technical"
```
Comment (prince-chrismc) [inline recipes/libsmacker/all/conanfile.py:9]:
```suggestion
    homepage = "http://libsmacker.sourceforge.net"
```

**Review by prince-chrismc — APPROVED** [review]

File: recipes/libsmacker/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+from contextlib import contextmanager
+import os
+
+
+class LibsmackerConan(ConanFile):
+    name = "libsmacker"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "http://libsmacker.sourceforge.net"
+    topics = ("conan", "libsmacker", "decoding ", "smk", "smacker", "video", "file")
+    license = "LGPL-v2.1"
```
Comment (uilianries) [inline recipes/libsmacker/all/conanfile.py:11]:
```suggestion
    license = "LGPL-2.1-or-later"
```

https://spdx.org/licenses/LGPL-2.1-or-later.html

File: recipes/libsmacker/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
+from conans import AutoToolsBuildEnvironment, ConanFile, tools
+from contextlib import contextmanager
+import os
+
+
```
Comment (SpaceIm) [inline recipes/libsmacker/all/conanfile.py:5]:
```suggestion
import os

required_conan_version = ">=1.29.1"


```

due to `tools.remove_files_by_mak()`

File: recipes/libsmacker/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+            "--enable-static={}".format(yes_no(not self.options.shared)),
+        ]
+        self._autotools.configure(configure_dir=self._source_subfolder, args=args)
+        return self._autotools
+
+    def build(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+        with tools.chdir(self._source_subfolder):
+            self.run("autoreconf -fiv", win_bash=tools.os_info.is_windows)
```
Comment (SpaceIm) [inline recipes/libsmacker/all/conanfile.py:82]: What about `self.run("{} -fiv".format(tools.get_env("AUTORECONF")), win_bash=tools.os_info.is_windows)` instead?

Comment (madebr) [reply recipes/libsmacker/all/conanfile.py:82]: The change is ok, although is is very unlikely that the name of the `autoreconf` script will change in the future.

Overriding autoreconf by setting `AUTORECONF` in a profile, won't work either way, I think.
Will the environment variables of the automake build requirement overrule?

```suggestion
            self.run("{} -fiv".format(tools.get_env("AUTORECONF")), win_bash=tools.os_info.is_windows)
```

**Review by SpaceIm — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**